File: 프로그래머스/0/181895. 배열 만들기 3/배열 만들기 3.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int[][] intervals) {
        ArrayList<Integer> answer = new ArrayList<>();
        
        for(int[] interval : intervals) {
            for( int i = interval[0]; i <= interval[1]; i++ ) {
                answer.add(arr[i]);        
            }
        }
        
        return answer.stream()
                     .mapToInt(Integer::intValue)
                     .toArray();
    }
}

// 다른 풀이
return Arrays.stream(intervals)
             .flatMapToInt(ints -> Arrays.stream(Arrays.copyOfRange(arr, ints[0], ints[1] + 1))).toArray();

File: 프로그래머스/0/181895. 배열 만들기 3/배열 만들기 3.py
def solution(arr, intervals):

    return arr[intervals[0][0]:intervals[0][1] + 1] + arr[intervals[1][0]:intervals[1][1] + 1]

# 다른 풀이
s1, e1 = intervals[0]
s2, e2 = intervals[1]
return arr[s1:e1+1] + arr[s2:e2+1]

# 다른 풀이
answer = []
for a,b in intervals: answer+=arr[a:b+1]
return answer
